srml/babe/src/lib.rs

```diff
@@ -152,6 +158,52 @@ decl_storage! {
 		/// Randomness under construction.
 		UnderConstruction: [u8; 32 /* VRF_OUTPUT_LENGTH */];
 	}
+	add_extra_genesis {
+		config(authorities): Vec<(AuthorityId, BabeWeight)>;
+		build(|
+			storage: &mut sr_primitives::StorageOverlay,
+			_: &mut sr_primitives::ChildrenStorageOverlay,
+			config: &GenesisConfig,
+		| {
+			runtime_io::with_storage(storage, || {
+				let mut authorities = config.authorities.clone();
+
+				// if session module is here AND it has mutated initial validators list, then we want to sync
```

Comment:
This code is written in assumption that `config.babe.authorities` is the subset of `config.session.keys`. If that isn't always true, this should be fixed.

Reply:
This wasn't true for `node-executor` tests. I'm not sure - if it is a test problem, or my assumption is invalid. For now I've fixed tests.